homeassistant/components/rfxtrx/device_action.py

```diff
@@ -0,0 +1,86 @@
+"""Provides device automations for RFXCOM RFXtrx."""
+from __future__ import annotations
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_DEVICE_ID, CONF_DOMAIN, CONF_TYPE
+from homeassistant.core import Context, HomeAssistant
+import homeassistant.helpers.config_validation as cv
+
+from . import DATA_RFXOBJECT, DOMAIN
+from .helpers import async_get_device_object
+
+CONF_DATA = "data"
+CONF_SUBTYPE = "subtype"
+
+ACTION_TYPE_COMMAND = "send_command"
+ACTION_TYPE_STATUS = "send_status"
+
+ACTION_TYPES = {
+    ACTION_TYPE_COMMAND,
+    ACTION_TYPE_STATUS,
+}
+
+ACTION_SELECTION = {
+    ACTION_TYPE_COMMAND: "COMMANDS",
+    ACTION_TYPE_STATUS: "STATUS",
+}
+
+ACTION_SCHEMA = cv.DEVICE_ACTION_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_TYPE): vol.In(ACTION_TYPES),
+        vol.Required(CONF_SUBTYPE): str,
```

Comment:
How is the subtype used?
It's referenced neither in  `async_get_action_capabilities` nor in `async_call_action_from_config`

Reply:
It's mostly used for display purposes. There is a direct correlation between data and the subtype value. I could potentially allow the data field to be either an INT or a STR value. Then if it's a str value in the call step. do the translation to int in that step instead of at generation time.

So SUBTYPE is used during config and display. Data is used when we execute.

Reply:
I've dropped the data part. Now we only have subtype.